rest/provider.go

```diff
@@ -164,6 +166,32 @@ func (p *Provider) Configure(ctx context.Context, req *pulumirpc.ConfigureReques
 		return resp, err
 	}
 
+	// Override the API host, if required. Intended for providers where the server names in the
+	// openapi spec will not match the API host that the provider needs to interact with during a deployment.
+	// To set via pulumi config, this will be "providername:apiHost"
+	// Otherwise, this can be set via the API_HOST env var
+	apiHost, ok := req.GetVariables()[fmt.Sprintf("%s:config:apiHost", p.name)]
```

Comment:
I'm not sure this configuration key value is right - can we assume the `p.name` will be the prefix?